packages/app/components/stack-modal/StackModal.tsx

```diff
@@ -32,100 +31,226 @@ import {
+              {stackIsComplete
+                ? "Complete"
+                : formatTimestampToDateWithTime(nextSlot)}
+            </StackDetail>
           </div>
-          <Button
-            action="quaternary"
-            iconLeft="close"
-            size="icon"
-            onClick={closeAction}
-          />
-        </div>
-      </ModalHeader>
-      <ModalContent className="px-0">
-        <div className="grid grid-cols-2 gap-5 px-4 gap-x-8 md:grid-cols-4">
-          <StackDetail title="Starts on">
-            {formatTimestampToDateWithTime(firstSlot)}
-          </StackDetail>
-          <StackDetail title="Ends on">
-            {formatTimestampToDateWithTime(lastSlot)}
-          </StackDetail>
-          <StackDetail title="Frequency">
-            Every {formatFrequencyHours(Number(stackOrder.interval))}
-          </StackDetail>
-          <StackDetail title="Next order">
-            {stackIsComplete
-              ? "Complete"
-              : formatTimestampToDateWithTime(nextSlot)}
-          </StackDetail>
-        </div>
-        <div className="w-full my-4 border-b border-surface-50"></div>
-        <div className="px-4">
-          <TitleText size={2} weight="bold">
-            Orders
-          </TitleText>
-          <StackProgress stackOrder={stackOrder} />
-          <StackInfo stackOrder={stackOrder} />
-          {totalStackOrdersDone(stackOrder) > 0 && (
-            <StackOrdersTable stackOrder={stackOrder} />
+          <div className="w-full my-4 border-b border-surface-50"></div>
+          <div className="px-4">
+            <TitleText size={2} weight="bold">
+              Orders
+            </TitleText>
+            <StackProgress stackOrder={stackOrder} />
+            <StackInfo stackOrder={stackOrder} />
+            {totalStackOrdersDone(stackOrder) > 0 && (
+              <StackOrdersTable stackOrder={stackOrder} />
+            )}
+          </div>
+        </ModalContent>
+        <ModalFooter>
+          {!stackOrder.cancelledAt && !stackIsComplete && (
+            <Button
+              size="sm"
+              action="secondary"
+              onClick={() => setIsOpenCancellationDialog(true)}
+              width="full"
+            >
+              Cancel Stacking
+            </Button>
           )}
-        </div>
-      </ModalContent>
-      <ModalFooter>
-        {!stackIsComplete && (
-          <Button
-            size="sm"
-            action="secondary"
-            onClick={() => alert("Are you sure you want to cancel stacking?")}
-            width="full"
-          >
-            Cancel Stacking
-          </Button>
+        </ModalFooter>
+      </Modal>
+      <Dialog
+        isOpen={isOpenCancellationDialog}
+        closeAction={() => setIsOpenCancellationDialog(false)}
+      >
+        <DialogContent
+          title=" Are you sure you want to cancel stacking?"
+          description={`The ${remainingFundsDescription}, will be sent to your wallet`}
```

Comment:
Just suggesting better punctuation.

```suggestion
          description={`The ${remainingFundsDescription} will be sent to your wallet.`}
```

Reply:
done